acestep/gradio_ui/events/training_handlers.py

```
@@ -826,4 +1031,184 @@ def export_lora(
+        import torch
+
+        # Even when exporting from final/, try to enrich metadata from the latest
+        # checkpoint training_state (epoch/step/run metadata), if available.
+        if latest_checkpoint is None and os.path.exists(checkpoint_dir):
+            checkpoints = [d for d in os.listdir(checkpoint_dir) if d.startswith("epoch_")]
+            if checkpoints:
+                checkpoints.sort(key=lambda x: int(x.split("_")[1]))
+                latest_checkpoint = checkpoints[-1]
+                try:
+                    latest_epoch = int(latest_checkpoint.split("_")[1])
+                except Exception:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)